WFLY-21223 Upgrade com.h2database:h2 from 2.2.224 to 2.4.240 in preview

rhusar opened the pull request:
Fix parameter duplication in testsuite/pom.xml

Resolves
https://issues.redhat.com/browse/WFLY-21223

rhusar committed:
WFLY-21223 Upgrade com.h2database:h2 from 2.2.224 to 2.4.240
Add missing node about idiosyncrasies of updating H2 in wf-full
Fix parameter duplication in testsuite/pom.xml
Fix names in JCA tests

Signed-off-by: Radoslav Husar <radosoft@gmail.com>

bstansberry commented:
Our H2 dep is driven by what the ORM version we integrate uses/tests as part of testing its H2 dialect. So this should come into WF Preview along with the upgrade to ORM 7.2, which uses it. ORM 6.6 still use 2.2.224.

bstansberry commented:
ORM has adapted to newer versions. ORM 6.6 is a maintenance branch.

Standard WildFly is somewhat behind the ecosystem as we wait to complete EE 11 compliance and catch up. (ORM 7.x is a Jakarta Persistence 3.2 / EE 11 impl.) The sooner we get std WF to EE 11 the better.

Fair point about the pom.xml not having a comment by the version prop discussing how managed that dep.

rhusar committed:
Only upgrade "WFLY-21223 Upgrade com.h2database:h2 to 2.4.240" in preview which is ORM 7.x

This commit is revertable, reverting this will upgrade the version in standard distro.

Signed-off-by: Radoslav Husar <radosoft@gmail.com>

rhusar commented:
@bstansberry OK, thanks for pointing out and discussing the idiosyncrasies of H2 version management in standard vs preview. I have added comments in both places so nobody has to go through these hoops again.

I have structured this change set to two commits where the 2nd commit is revertable; reverting of which will upgrade this properly in standard and pull out of preview.

bstansberry commented on testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jca/metrics/data-sources/complex-driver.xml:
```diff
@@ -6,7 +6,7 @@
     <datasources>
         <drivers>
-            <driver name="name" module="com.h2database.h2" major-version="2" minor-version="4">
+            <driver name="name" module="com.h2database.h2" major-version="2" minor-version="2">
```
We'll probably have to use an expression for this and drive the system property based on what's being tested.

Unless the uses of this config aren't really deriving any value from setting this, in which case we can just remove this. AIUI these two version attributes are used to compare the config against what the driver itself reports, failing if there's a mismatch. I assume that's why the test fails in the WildFly Preview job. If an attribute isn't set it doesn't do the comparison. I don't see any other use.
